### PR TITLE
SplitButton: Typescript migration

### PR DESCRIPTION
**Pull request by SergeyRoyt**
https://monday.monday.com/boards/2861766393/pulses/3390016536

### CODE REVIEW

File: src/components/SplitButton/SplitButton.tsx [hunk 1]
```diff
@@ -1,69 +1,94 @@
...
+    className,
+    leftIcon,
+    rightIcon,
+    onClick,
+    children,
+    marginLeft,
+    marginRight
+  } = overrideButtonProps;
+
+  console.log("Button.defaultProps", Button.defaultProps);
```
Comment (hadasfa) [inline src/components/SplitButton/SplitButton.tsx:88]: let's remove console.logs ?

File: src/components/SplitButton/SplitButton.tsx [hunk 2]
```diff
@@ -1,69 +1,94 @@
...
+  zIndex = null,
   secondaryDialogClassName,
-  secondaryDialogPosition,
-  dialogPaddingSize,
+  secondaryDialogPosition = SplitButtonSecondaryContentPosition.BOTTOM_START,
+  // @ts-ignore TODO TS-migration, when DialogContentContainer is converted to TS
+  dialogPaddingSize = DialogContentContainer.sizes.MEDIUM,
   ...buttonProps
 }) => {
+  const overrideButtonProps = { ...Button.defaultProps, ...buttonProps };
```
Comment (hadasfa) [inline src/components/SplitButton/SplitButton.tsx:72]: Why we need Button.defaultProps? when we pass it to button, the button default props will override undefined values?

**Review by hadasfa — CHANGES_REQUESTED** [review]
Few small notes

File: src/components/SplitButton/SplitButton.tsx (hunk 2 above)
Comment (SergeyRoyt) [reply src/components/SplitButton/SplitButton.tsx:72]: Yes, but these button props are used to calculate props and classes for SplitButton as well - not only for Button. That's why we have to know their values before passing them to Button

**Review by orrgottlieb — APPROVED** [review]

**Pull request merged**